### PR TITLE
Update nncf to 2.0

### PR DESCRIPTION
**Pull request by daniil-lyakhov**
Pending training to check functionality:

- [x] Efficientnet bo int8
- [x] Mobilenet v3 large 1 int8

### CODE REVIEW

File: torchreid/engine/engine.py [hunk 1]
```diff
@@ -372,16 +374,26 @@ def run(
...
                 perf_monitor=perf_monitor,
                 stop_callback=stop_callback,
+                compression_ctrl=compression_ctrl
             )
+            if compression_ctrl is not None:
+                print(compression_ctrl.statistics().to_str())
+
             if stop_callback and stop_callback.check_stop():
-                break
+                if compression_ctrl.compression_stage == CompressionStage.FULLY_COMPRESSED:
```
Comment (sovrasov) [inline torchreid/engine/engine.py]: `stop_callback` is passed to the run() only in sonoma creek environment, so this statement has no effect if the training is launched via `main.py`

Comment (daniil-lyakhov) [reply torchreid/engine/engine.py]: Should I do the same here? https://github.com/openvinotoolkit/deep-object-reid/pull/62/files#diff-5b9e00f3cdc0923e0eac49367d72bf27e52682038f424a27d35c3472b2f2cb37R427-R434

File: torchreid/engine/engine.py [hunk 2]
```diff
@@ -419,7 +431,11 @@ def run(
                         self.save_model(self.epoch, save_dir, is_best=is_candidate_for_best,
                                             should_save_ema_model=should_save_ema_model)
                     if should_exit:
-                        break
+                        if compression_ctrl.compression_stage == CompressionStage.FULLY_COMPRESSED:
+                            break
+                        else:
+                            print('Try to early exit training but skip due to compression algo didn\'t'
```
Comment (kprokofi) [inline torchreid/engine/engine.py]: I guess we should keep it silently for users and do not print anything. The training hasn't been finished yet and okay

File: torchreid/engine/engine.py [hunk 3]
```diff
@@ -778,6 +797,8 @@ def _feature_extraction(data_loader):
                 if self.use_gpu:
                     imgs = imgs.cuda()
 
+                # TODO(kprokofi) check features is tuple
```
Comment (kprokofi) [inline torchreid/engine/engine.py]: This branch isn't related to classification. if we ever get to check the functionality of this subtask, then we will fix what does not work

File: torchreid/engine/image/am_softmax.py [hunk 4]
```diff
@@ -262,10 +263,14 @@ def forward_backward(self, data):
 
                 total_loss += coeff_mutual_learning * mutual_loss / float(num_mutual_losses)
 
-
             total_loss.backward(retain_graph=self.enable_metric_losses)
 
             for model_name in model_names:
+                # TODO(kshpv, lbeynens): Change this ugly code. The bug appears trying to
```
Comment (kprokofi) [inline torchreid/engine/image/am_softmax.py]: this TODO  has been here for a long time

File: torchreid/engine/engine.py (hunk 1 above)
Comment (kprokofi) [reply torchreid/engine/engine.py]: I think we may just delete this condition. If the callback to stop passes from Sonoma Сreek SDK, then we don't mean to interfere with this

**Comment by kprokofi** [comment]
@daniil-lyakhov , where is the EMA fix?

File: torchreid/engine/engine.py (hunk 3 above)
Comment (daniil-lyakhov) [reply torchreid/engine/engine.py]: Ok, I thought it will be useful in future. I'll remove it

File: torchreid/engine/image/am_softmax.py (hunk 4 above)
Comment (daniil-lyakhov) [reply torchreid/engine/image/am_softmax.py]: I've changed it, please check

File: torchreid/engine/engine.py (hunk 3 above)
Comment (daniil-lyakhov) [reply torchreid/engine/engine.py]: Done

File: torchreid/engine/engine.py (hunk 2 above)
Comment (daniil-lyakhov) [reply torchreid/engine/engine.py]: Done

File: torchreid/engine/engine.py (hunk 1 above)
Comment (daniil-lyakhov) [reply torchreid/engine/engine.py]: Done

File: torchreid/engine/engine.py [hunk 5]
```diff
@@ -433,9 +430,6 @@ def run(
                     if should_exit:
                         if compression_ctrl.compression_stage == CompressionStage.FULLY_COMPRESSED:
```
Comment (kprokofi) [inline torchreid/engine/engine.py]: when the normal training is going without NNCF, this method returns true?

File: torchreid/engine/image/am_softmax.py [hunk 6]
```diff
@@ -59,7 +59,7 @@ def __init__(self, datamanager, models, optimizers, reg_cfg, metric_cfg, schedul
                                                    use_ema_decay=use_ema_decay,
                                                    ema_decay=ema_decay)
 
-        assert softmax_type in ['softmax', 'am', 'asl']
+        assert softmax_type in ['softmax', 'am_softmax', 'asl']
         assert s > 0.0
         if softmax_type == 'am':
```
Comment (kprokofi) [inline torchreid/engine/image/am_softmax.py]: this 'am' also must be 'am_softmax'

File: torchreid/engine/image/am_softmax.py (hunk 4 above)
Comment (kprokofi) [reply torchreid/engine/image/am_softmax.py]: I mean, if somebody is going to actually do this TODO? =)

File: torchreid/engine/engine.py (hunk 5 above)
Comment (daniil-lyakhov) [reply torchreid/engine/engine.py]: No, I've checked if compression_ctrl is not None for now, please check

File: torchreid/engine/image/am_softmax.py (hunk 6 above)
Comment (daniil-lyakhov) [reply torchreid/engine/image/am_softmax.py]: Done

**Review by alexsu52 — APPROVED** [review]

**Comment by alexsu52** [comment]
@sovrasov, @kprokofi Please merge if you have no comments.

**Review by sovrasov — APPROVED** [review]

**Pull request merged**